File: src/main/java/com/dainxt/weaponthrow/mixins/PlayerEntityMixin.java
package com.dainxt.weaponthrow.mixins;

import net.minecraft.entity.player.PlayerEntity;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.Unique;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import com.dainxt.weaponthrow.Interface.IPlayerEntityMixin;
import com.dainxt.weaponthrow.capabilities.PlayerThrowData;
import com.dainxt.weaponthrow.events.OnStartPlayerTick;

@Mixin(PlayerEntity.class)
public class PlayerEntityMixin implements IPlayerEntityMixin {
    @Unique
    private PlayerThrowData throwPower = new PlayerThrowData((PlayerEntity)(Object)this);

    @Override
    public void weaponThrow$setThrowPower(PlayerThrowData value) {
        throwPower = value;
    }

    @Override
    public PlayerThrowData weaponThrow$getThrowPower() {
        return throwPower;
    }

    @Inject(method = "tick", at = @At("HEAD"))
    private void init(CallbackInfo info) {
        OnStartPlayerTick.EVENT.invoker().interact((PlayerEntity)(Object)this);
    }
}

File: src/main/java/com/dainxt/weaponthrow/mixins/PlayerEntityMixin.kt
package com.dainxt.weaponthrow.mixins

import com.dainxt.weaponthrow.capabilities.PlayerThrowData
import com.dainxt.weaponthrow.events.OnStartPlayerTick
import com.dainxt.weaponthrow.interfaces.IPlayerEntityMixin

import net.minecraft.entity.player.PlayerEntity
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(PlayerEntity::class)
abstract class PlayerEntityMixin: IPlayerEntityMixin {
	private var throwPower: PlayerThrowData = PlayerThrowData(this as Any as PlayerEntity)

	override fun setThrowPower(value: PlayerThrowData) {
		throwPower = value
	}

	override fun getThrowPower(): PlayerThrowData {
		return throwPower
	}

	@Inject(at = [At("HEAD")], method = ["tick"])
	fun init(info: CallbackInfo) {
//		throwPower =
		OnStartPlayerTick.EVENT.invoker().interact(this as Any as PlayerEntity)

//		println("Interface Path is ${IPlayerEntityMixin::class.java.`package`.name}")
	}
}